core/src/test/scala/caliban/execution/FragmentSpec.scala

```diff
@@ -50,6 +51,35 @@ object FragmentSpec extends ZIOSpecDefault {
           res <- int.execute(query)
         } yield assertTrue(res.data.toString == """{"amos":{"name":"Amos Burton"}}""")
       },
+      test("fragment with inner and outer") {
+        implicit lazy val bazSchema: Schema[Any, FragmentSchema.Baz] = Schema.gen
+
+        val interpreter = graphQL(FragmentSchema.resolverFooBar).interpreter
+        val query       = gqldoc("""
+           query {
+             bar {
+               ...Outer
+             }
+           }
+           fragment Outer on Bar {
+             baz {
+               foo {
+                id
+               }
+               ...Inner
+             }
+           }
+           fragment Inner on Baz {
+             foo {
+               id
+            }
```

Comment:
Do we have tests for the failing case as well? Where the fragments don’t match exactly?

Reply:
Yep, we have 2: "inline fragment selection with different field types" and "conflicting different responses".